src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,167 @@
+package duke;
+
+import command.AddCommand;
+import command.Command;
+import command.DeleteCommand;
+import command.DoneCommand;
+import command.ExitCommand;
+import command.IncorrectCommand;
+import command.ListCommand;
+import command.FindCommand;
+
+import task.Deadline;
+import task.Event;
+import task.Todo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser is the class which makes sense of the user's commands.
+ *
+ * @author Joshua
+ */
+public class Parser {
+    private final static String COMMAND_LIST = "list";
+    private final static String COMMAND_TERMINATE = "bye";
+    private final static String COMMAND_COMPLETE_TASK = "done";
+    private final static String COMMAND_ADD_TODO = "todo";
+    private final static String COMMAND_ADD_DEADLINE = "deadline";
+    private final static String COMMAND_ADD_EVENT = "event";
+    private final static String COMMAND_DELETE_EVENT = "delete";
+    private final static String COMMAND_FIND = "find";
+    private final static String DATE_DEADLINE = "/by";
+    private final static String DATE_EVENT = "/at";
+
+    private final static DateTimeFormatter SAVE_READ_DATETIME_FORMAT = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
+
+    /**
+     * Creates a Parser object.
+     */
+    public Parser() {
+    }
+
+    /**
+     * Types of task that could result from the parsed user command.
+     */
+    enum TypeOfCommand {
+        DONE,
+        DEADLINE,
+        TODO,
+        EVENT,
+        DELETE,
+        FIND
+    }
+
+    /**
+     * Parses the input from the user and returns a command with respect to the
+     * input from the user.
+     *
+     * @param fullCommand the given input from the user.
+     * @return the command that will be carried out by Duke.
+     */
+    public static Command parse(String fullCommand) {
+        String shortCommand;
+        String restOfCommand;
+        if (fullCommand.contains(" ")) {
+            shortCommand = fullCommand.substring(0, fullCommand.indexOf(' '));
+            restOfCommand = fullCommand.substring(shortCommand.length() + 1);
+        } else {
+            shortCommand = fullCommand;
+            restOfCommand = "";
+        }
+        switch (shortCommand) {
```

Comment:
Should there be breaks for each case? Or, if break statements were left out intentionally, should there be explicit //Fallthrough comments?

Reply:
My IDE is warning against adding break statements after my return lines and I think it is unnecessary as well because the code would terminate already if it reaches the return statement. Thanks for noticing that though!